### PR TITLE
relax package requirements to <= type conditions and support latest mxnet …

### PR DESCRIPTION
**Pull request by TammoR**
Support more flexible requirements for pip installing in different deployment settings.

### CODE REVIEW

File: requirements/requirements.dev.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
-pylint==2.1.1
-typing==3.6.6
-mypy==0.620
-sphinx==1.7.6
-pillow==5.2.0
-pytest==3.7.3
+setuptools>=40.2.0
+wheel>=0.31.1
+pylint>=2.1.1
+ipython>=6.2.1
```
Comment (andrey-tpt) [inline requirements/requirements.dev.txt]: Why did you remove explicit dependency on Typing module? It's used in the code and it's unclear from dependency on iPython that it's used. Moreover, I'm concerned why do we have iPython here - is it required for build ?

File: requirements/requirements.gpu-cu91.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-numpy==1.15.0
-scikit-learn[alldeps]==0.19.0
-pandas==0.22.0
-mxnet-cu91==1.3.0b20180820
+numpy>=1.15.0
```
Comment (andrey-tpt) [inline requirements/requirements.gpu-cu91.txt]: You have numpy 1.13.1 everywhere except this place. Why ?

File: requirements/requirements.txt [hunk 3]
```diff
@@ -1,5 +1,4 @@
-numpy==1.15.0
-scikit-learn[alldeps]==0.19.0
-pytest==3.2.1
-pandas==0.22.0
-mxnet==1.3.0b20180820
\ No newline at end of file
+numpy>=1.13.1
+scikit-learn[alldeps]>=0.18.1
+pandas>=0.22.0
+mxnet>=1.0.0
```
Comment (andrey-tpt) [inline requirements/requirements.txt]: Regarding mxnet. It's known that datawig doesn't work (tests crash) with versions of mxnet < 1.3

File: requirements/requirements.gpu-cu91.txt (hunk 2 above)
Comment (tdhd) [inline requirements/requirements.gpu-cu91.txt]: does the newer mxnet version require more recent numpy?
